src/main/kotlin/no/nav/familie/tilbake/config/KafkaErrorHandler.kt

```diff
@@ -0,0 +1,89 @@
+import org.springframework.kafka.listener.ContainerStoppingErrorHandler
+import org.springframework.kafka.listener.MessageListenerContainer
+import org.springframework.stereotype.Component
+import java.time.Duration
+import java.util.concurrent.Executor
+import java.util.concurrent.atomic.AtomicInteger
+import java.util.concurrent.atomic.AtomicLong
+
+@Component
+class KafkaErrorHandler : ContainerStoppingErrorHandler() {
+
+    val LOGGER: Logger = LoggerFactory.getLogger(KafkaErrorHandler::class.java)
+    val SECURE_LOGGER: Logger = LoggerFactory.getLogger("secureLogger")
+
+    private val executor: Executor
+    private val counter = AtomicInteger(0)
+    private val lastError = AtomicLong(0)
+    override fun handle(e: Exception,
+                        records: List<ConsumerRecord<*, *>>?,
+                        consumer: Consumer<*, *>,
+                        container: MessageListenerContainer) {
+        Thread.sleep(1000)
+
+        if (records.isNullOrEmpty()) {
+            LOGGER.error("Feil ved konsumering av melding. Ingen records. ${consumer.subscription()}", e)
+            scheduleRestart(e,
+                            records,
+                            consumer,
+                            container,
+                            "Ukjent topic")
+        } else {
+            records.first().run {
+                LOGGER.error("Feil ved konsumering av melding fra ${this.topic()}. id ${this.key()}, " +
+                             "offset: ${this.offset()}, partition: ${this.partition()}")
+                SECURE_LOGGER.error("${this.topic()} - Problemer med prosessering av $records", e)
+                scheduleRestart(e,
+                                records,
+                                consumer,
+                                container,
+                                this.topic())
+            }
+        }
+    }
+
+    private fun scheduleRestart(e: Exception,
+                                records: List<ConsumerRecord<*, *>>? = null,
+                                consumer: Consumer<*, *>,
+                                container: MessageListenerContainer,
+                                topic: String) {
+        val now = System.currentTimeMillis()
+        if (now - lastError.getAndSet(now) > COUNTER_RESET_TIME) {
+            counter.set(0)
+        }
+        val numErrors = counter.incrementAndGet()
+        val stopTime =
+                if (numErrors > SLOW_ERROR_COUNT) LONG else SHORT * numErrors
+        executor.execute {
+            try {
+                Thread.sleep(stopTime)
+                LOGGER.warn("Starter kafka container for {}", topic)
+                container.start()
+            } catch (exception: Exception) {
+                LOGGER.error("Feil oppstod ved venting og oppstart av kafka container", exception)
+            }
+        }
+        LOGGER.warn("Stopper kafka container for {} i {}", topic, Duration.ofMillis(stopTime).toString())
```

Comment:
logger kan endres til "Stopper kafka container for $topic i ${Duration.ofMillis(stopTime).toString()}"